build.sbt

```diff
@@ -103,53 +105,77 @@ lazy val commonSettings = Seq(
   // format: on
 )
 
-// add support for Scala version ranges such as "scala-2.12+" in source folders (single version folders such as
-// "scala-2.11" are natively supported by SBT).
-// In order to keep this simple, we're doing this case by case, taking advantage of the fact that we intend to support
-// only 3 major versions at any given moment.
+// add support for Scala version ranges such as "scala-2.12+" or "scala-2.13-" in source folders (single version folders
+// such as "scala-2.11" are natively supported by SBT).
 def crossVersionSharedSources(unmanagedSrcs: SettingKey[Seq[File]]) = {
   unmanagedSrcs ++= {
-    val minor = CrossVersion.partialVersion(scalaVersion.value).map(_._2)
-    List(
-      if (minor.exists(_ <= 12)) unmanagedSrcs.value.map { dir => new File(dir.getPath + "-2.12-") }
-      else Nil,
-      if (minor.exists(_ >= 12)) unmanagedSrcs.value.map { dir => new File(dir.getPath + "-2.12+") }
-      else Nil
-    ).flatten
+    val versionNumber = CrossVersion.partialVersion(scalaVersion.value)
+    val expectedVersions = Seq(scala211, scala212, scala213, scala30).flatMap(CrossVersion.partialVersion)
+    expectedVersions.flatMap { case v @ (major, minor) =>
+      List(
+        if (versionNumber.exists(Ordering[(Long, Long)].lteq(_, v)))
```

Comment:
nit: if you import `scala.math.Ordering.Implicits._` you can simply do `.exists (_ <= v)` and `.exists(_ >= v)` here.